wordpress-mobile/WordPress-iOS
# Blogging Prompts: Add support for Bloganuary field

Resolves #22026 and #22027
Depends on #21983 

This PR adds the capability to parse Bloganuary fields into the `BloggingPrompt` model. A summary of the changes:

- The `bloganuary_id` field is optionally parsed in `BloggingPromptRemoteObject`.
- This property is then abstracted through a `BrandContext` enum, where it assigns context-specific data to the `BloggingPrompt` model. The idea is to make the `BloggingPrompt` model "agnostic" about any specific events.
- The Bloganuary icon is added, and it should adjust properly in dark and light mode.
- When answering Bloganuary prompts, the additional tags should be appended to the post.

## To test

To make testing easier, in `BloggingPromptRemoteObject`'s `init` method (line 70), add this temp code so that `bloganuary_id` always exists:

```swift
        self.bloganuaryId = {
                return "bloganuary-2024-01" // add this line
```

With the above change in place:

- Launch the Jetpack app.
- Log in with a WP.com account.
- 🔎  Verify that the Bloganuary attribution is shown. 
- Switch to the other color scheme.
- 🔎 Verify that the Bloganuary icon is updated properly.
- Tap "Answer Prompt".
- From the Post Editor, tap the ellipsis menu and select Post Settings.
- 🔎  Verify that additional tags `bloganuary` and `bloganuary-2024-01` are added.

Next, remove the test code above to return it to the original state to test non-Bloganuary prompts.

- Launch the Jetpack app again.
- 🔎  Verify that the Bloganuary attribution is NOT shown.
- Tap "Answer Prompt".
- From the Post Editor, tap the ellipsis menu and select Post Settings.
- 🔎  Verify that only `dailyprompt` and `dailyprompt-<id>` tags are present.

## Regression Notes
1. Potential unintended areas of impact
Non-Bloganuary prompts may be affected.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manually tested the changes.

3. What automated tests I added (or what prevented me from doing so)
Added unit tests.

PR submission checklist:


UI Changes testing checklist:
